### PR TITLE
Add tag for xgboost 1.6.2

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by adamjstewart** [comment]
Can you also update `py-xgboost`?

**Comment by iarspider** [comment]
> Can you also update `py-xgboost`?

@adamjstewart done

File: var/spack/repos/builtin/packages/py-xgboost/package.py [hunk 1]
```diff
@@ -29,10 +30,11 @@ class PyXgboost(PythonPackage):
     variant("dask", default=False, description="Enables Dask extensions for distributed training.")
     variant("plotting", default=False, description="Enables tree and importance plotting.")
 
-    for ver in ["1.3.3", "1.5.2", "1.6.1"]:
+    for ver in ["1.3.3", "1.5.2", "1.6.1", "1.6.2"]:
         depends_on("xgboost@" + ver, when="@" + ver)
 
-    depends_on("python@3.7:", when="@1.6:", type=("build", "run"))
+    depends_on("python@3.8:", when="@1.6.2:", type=("build", "run"))
+    depends_on("python@3.7:", when="@1.6.1", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xgboost/package.py]: Seems like 3.7+ is still supported in 1.6.2, I would undo this change

File: var/spack/repos/builtin/packages/py-xgboost/package.py [hunk 2]
```diff
@@ -29,10 +30,10 @@ class PyXgboost(PythonPackage):
     variant("dask", default=False, description="Enables Dask extensions for distributed training.")
     variant("plotting", default=False, description="Enables tree and importance plotting.")
 
-    for ver in ["1.3.3", "1.5.2", "1.6.1"]:
+    for ver in ["1.3.3", "1.5.2", "1.6.1", "1.6.2"]:
         depends_on("xgboost@" + ver, when="@" + ver)
 
-    depends_on("python@3.7:", when="@1.6:", type=("build", "run"))
+    depends_on("python@3.7:", when="@1.6.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xgboost/package.py]:
```suggestion
    depends_on("python@3.7:", when="@1.6:", type=("build", "run"))
```
1.6.0 also requires 3.7+

**Review by adamjstewart — APPROVED** [review]
